pageserver/src/tenant/delta_layer.rs

```diff
@@ -768,6 +768,57 @@ impl DeltaLayerWriter {
     }
 }
 
+pub struct DeltaLayerWriter {
+    inner: Option<DeltaLayerWriterInner>,
+}
+
+impl DeltaLayerWriter {
+    pub fn new(
+        conf: &'static PageServerConf,
+        timeline_id: TimelineId,
+        tenant_id: TenantId,
+        key_start: Key,
+        lsn_range: Range<Lsn>,
+    ) -> Result<Self> {
+        Ok(Self {
+            inner: Some(DeltaLayerWriterInner::new(conf, timeline_id, tenant_id, key_start, lsn_range)?),
+        })
+    }
+
+    pub fn put_value(&mut self, key: Key, lsn: Lsn, val: Value) -> Result<()> {
+        self.inner.as_mut().unwrap().put_value(key, lsn, val)
+    }
+
+    pub fn put_value_bytes(
+        &mut self,
+        key: Key,
+        lsn: Lsn,
+        val: &[u8],
+        will_init: bool,
+    ) -> Result<()> {
+        self.inner.as_mut().unwrap().put_value_bytes(key, lsn, val, will_init)
+    }
+
+    pub fn size(&self) -> u64 {
+        self.inner.as_ref().unwrap().size()
+    }
+
+    pub fn finish(mut self, key_end: Key) -> Result<DeltaLayer> {
+        self.inner.take().unwrap().finish(key_end)
+    }
+}
+
+impl Drop for DeltaLayerWriter {
```

Comment:
Please add a comment here, to explain what this is for. (To remove the incomplete file if the caller drops the writer without calling `finish`)